### PR TITLE
Feat 219 keymap cheatsheet

### PR DESCRIPTION
**Pull request by adamrocker**
New feature for generating the keymap cheat sheet with PDF format.
スクリーンショット 2021-03-02 0 20 44
スクリーンショット 2021-03-02 0 20 51

This feature is generating the every layer keymap which are displayed currently.
Both of the key label's language and layer option are also applied to the cheat sheet.
スクリーンショット 2021-03-02 0 24 20

スクリーンショット 2021-03-02 0 28 02

スクリーンショット 2021-03-02 0 28 13

### CODE REVIEW

File: src/components/configure/configuration/ConfigurationDialog.tsx [hunk 1]
```diff
@@ -209,6 +212,28 @@ export default class ConfigurationDialog extends React.Component<
...
+    }
+
+    const pdf = new KeymapPdfGenerator(
+      this.props.keyboardDefinition!.layouts.keymap,
+      keys,
+      this.props.layerCount!,
+      'en-gb'
+    );
+    pdf.genPdf(this.props.productName!);
+  }
```
Comment (yoichiro) [inline src/components/configure/configuration/ConfigurationDialog.tsx:235]: @adamrocker I guess that the purpose of this code is for testing during your development, therefore this method should be removed.

File: src/components/configure/configuration/ConfigurationDialog.tsx [hunk 2]
```diff
@@ -241,6 +266,13 @@ export default class ConfigurationDialog extends React.Component<
       >
         <DialogTitle id="draggable-dialog-title" style={{ cursor: 'move' }}>
           Configuration
+          <Button
+            onClick={() => {
+              this.onClickDownloadCheatSheet();
+            }}
+          >
+            download
+          </Button>
```
Comment (yoichiro) [inline src/components/configure/configuration/ConfigurationDialog.tsx:275]: @adamrocker It seems that you forgot this download button for testing. Could you remove this?

Screenshot from 2021-03-02 06-50-45

**Review by yoichiro — CHANGES_REQUESTED** [review]
@adamrocker I posted comments. Could you confirm them? Thanks!

File: src/components/configure/configuration/ConfigurationDialog.tsx (hunk 2 above)
Comment (adamrocker) [reply src/components/configure/configuration/ConfigurationDialog.tsx:275]: Sorry, this is a test button....　I forgot to remove. Let me clean up these test codes.

Comment (adamrocker) [reply src/components/configure/configuration/ConfigurationDialog.tsx]: @yoichiro I cleaned up all test codes in the dialog. Could you please confirm them again?

File: src/components/configure/configuration/ConfigurationDialog.tsx [hunk 3]
```diff
@@ -212,28 +209,6 @@ export default class ConfigurationDialog extends React.Component<
...
-
-    const pdf = new KeymapPdfGenerator(
-      this.props.keyboardDefinition!.layouts.keymap,
-      keys,
-      this.props.layerCount!,
-      'en-gb'
-    );
-    pdf.genPdf(this.props.productName!);
-  }
-
```
Comment (yoichiro) [inline src/components/configure/configuration/ConfigurationDialog.tsx]: :+1:

File: src/components/configure/configuration/ConfigurationDialog.tsx [hunk 4]
```diff
@@ -266,13 +241,6 @@ export default class ConfigurationDialog extends React.Component<
       >
         <DialogTitle id="draggable-dialog-title" style={{ cursor: 'move' }}>
           Configuration
-          <Button
-            onClick={() => {
-              this.onClickDownloadCheatSheet();
-            }}
-          >
-            download
-          </Button>
```
Comment (yoichiro) [inline src/components/configure/configuration/ConfigurationDialog.tsx]: :+1:

**Review by yoichiro — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**